src/time_series_utils.jl

```diff
@@ -11,6 +11,26 @@ const TIME_SERIES_STRING_TO_TYPE = Dict(
     "SingleTimeSeries" => SingleTimeSeries,
 )
 
+"""
+Parse time series type string to concrete type using static dispatch.
+This is more efficient for precompilation than dictionary lookup.
```

Comment:
It… replaced a dictionary lookup with a long if/else chain. That is not a style we want to encourage! If we're really concerned about allocations here, we might prototype whether a `Val` dispatch approach is better, but definitely not this.